src/pyeval.jl

```diff
@@ -202,32 +202,37 @@ pasted into the Python code.   This allows you to evaluate code
 where the code itself is generated by a Julia expression.
 """
 macro py_str(code, options...)
+    m = :(pynamespace($__module__))
+    esc(macroexpand(__module__, :(PyCall.@_py_str($m, $m, $code, $(options...)))))
+end
+
+macro _py_str(pyglobals, pylocals, code, options...)
```

Comment:
If we are going to use a function for this, it should take a named tuple `(__module__=__module__, __source__=__source__)` as the first argument so that all the information available within a usual macro is available.  I think it's better to make it a single first argument rather than two arguments because future versions of `julia` may introduce some more information within a macro definition but it'd be bad to change the calling convention of this function.

Reply:
Also, we should use `options` rather than `options...`.